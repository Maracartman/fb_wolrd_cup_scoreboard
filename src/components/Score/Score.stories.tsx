import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import '../../assets/fonts/Ozone/Ozone-xRRO.ttf';

import { Score } from './Score';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Example/Score',
  component: Score,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
  },
} as ComponentMeta<typeof Score>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Score> = (args) => <>
<style>
{
  `@font-face {
    font-family: "Ozone";
    src: local("Ozone"),
    url("./assets/fonts/Ozone/Ozone-xRRO.ttf") format("truetype");
    font-weight: normal;
  }
  `
}
</style>
<Score {...args} />
</>;

export const DefaultWithNoScore = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
DefaultWithNoScore.args = {
};

export const withScore = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
DefaultWithNoScore.args = {
  score: 12
};
