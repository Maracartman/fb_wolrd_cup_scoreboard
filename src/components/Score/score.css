.score {
  justify-content: center;
  display: flex;
  text-align: center;
}

.score-text {
  display: flex;
  border-radius: 10px;
  padding: 5vw;
  border: 3px dotted rgb(255, 255, 255);
  width: fit-content;
  font-size: 20vw;
  font-weight: 700;
  font-family: sans-serif;
}
