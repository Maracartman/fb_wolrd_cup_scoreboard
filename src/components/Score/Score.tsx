import React from 'react';
import { Box} from '@mui/material';
import './score.css';
import { transformToStringScore } from 'src/utils/string';

interface ScoreProps {
  score?: number
}

export const Score = ({
  score = 0
}: ScoreProps) => {
  const formattedScore = transformToStringScore(score)
  return (
    <Box className='score'>
     <div className='score-text'>
      {formattedScore}
     </div>
    </Box>
  );
};
