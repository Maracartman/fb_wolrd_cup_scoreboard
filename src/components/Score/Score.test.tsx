import React from 'react'
import {screen, render} from "@testing-library/react"
import {  Score } from './Score'


describe("Score component test suite", () => {
    it('match the snapshot', () => {
        const { container } = render(<Score />)

        expect(container).toMatchSnapshot()
    })

    it('display "00" if an undefined score is passed', () => { 
        render(<Score />)

        expect(screen.getByText('00')).toBeInTheDocument()
    })

    it('display 05 if the score 5 is passed to the Score component', () => {
        render(<Score score={5} />)

        expect(screen.getByText("05")).toBeInTheDocument()
    })
})