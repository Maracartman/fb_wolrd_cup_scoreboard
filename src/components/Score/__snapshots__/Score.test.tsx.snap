// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score component test suite match the snapshot 1`] = `
<div>
  <div
    class="score MuiBox-root css-0"
  >
    <div
      class="score-text"
    >
      00
    </div>
  </div>
</div>
`;
